Tasks/DownloadPipelineArtifactV2/task.json

```diff
@@ -0,0 +1,208 @@
+            "defaultValue": "latest",
+            "visibleRule": "buildType == specific",
+            "required": true,
+            "options": {
+                "latest": "Latest",
+                "latestFromBranch": "Latest from specific branch and specified Build Tags",
+                "specific": "Specific version"
+            }
+        },
+        {
+            "name": "branchName",
+            "type": "string",
+            "label": "Branch name",
+            "defaultValue": "refs/heads/master",
+            "visibleRule": "buildType == specific && buildVersionToDownload == latestFromBranch",
+            "required": true,
+            "helpMarkDown": "Specify to filter on branch/ref name, for example: ```refs/heads/develop```."
+        },
+        {
+            "name": "buildId",
+            "aliases": [
+                "pipelineId"
+            ],
+            "type": "pickList",
+            "label": "Build",
+            "defaultValue": "",
+            "required": true,
+            "visibleRule": "buildType == specific && buildVersionToDownload == specific",
+            "properties": {
+                "EditableOptions": "True",
+                "DisableManageLink": "True"
+            },
+            "helpMarkDown": "The build from which to download the artifacts."
+        },
+        {
+            "name": "tags",
+            "type": "string",
+            "label": "Build Tags",
+            "defaultValue": "",
+            "visibleRule": "buildType == specific && buildVersionToDownload != specific",
+            "required": false,
+            "helpMarkDown": "A comma-delimited list of tags. Only builds with these tags will be returned."
+        },
+        {
+            "name": "artifactName",
+            "type": "pickList",
+            "label": "Artifact name",
+            "defaultValue": "",
+            "required": false,
+            "properties": {
+                "EditableOptions": "True",
+                "DisableManageLink": "True"
+            },
+            "helpMarkDown": "The name of the artifact to download."
+        },
+        {
+            "name": "itemPattern",
+            "type": "multiLine",
+            "label": "Matching pattern",
+            "defaultValue": "**",
+            "required": false,
+            "properties": {
+                "rows": "3",
+                "resizable": "true"
+            },
+            "helpMarkDown": "Specify files to be downloaded as multi line minimatch pattern. [More Information](https://aka.ms/minimatchexamples).</p>"
+        },
+        {
+            "name": "downloadPath",
+            "aliases": [
+                "targetPath"
+            ],
+            "type": "string",
+            "label": "Destination directory",
+            "defaultValue": "$(Pipeline.Workspace)",
+            "required": true,
+            "helpMarkDown": "Path on the agent machine where the artifacts will be downloaded."
```

Comment:
`Directory to download the artifact files to. Can be relative to the pipeline workspace directory or absolute. If more than one artifact is selected for download, a sub-directory will be created for each.`